os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -208,56 +208,67 @@ static int rtk_drv_callback_handler(int argc, char *argv[])
 		break;
 	}
 
+	kmm_free(type);
+
 	return 0;
 }
 
 static void linkup_handler(rtk_reason_t *reason)
 {
-	//RTKDRV_ENTER;
-	pid_t pid;
-	char *argv[2];
-	argv[1] = NULL;
-	char data[2] = {0, 0};
+	pthread_t tid;
+	int ret;
+	int *type = (int *)kmm_malloc(sizeof(int));
+
+	if (type == NULL) {
+		vddbg("malloc error\n");
+		return;
+	}
 
 	if (g_mode == RTK_WIFI_STATION_IF) {
 		if (reason->reason_code == RTK_STATUS_SUCCESS) {
-			data[0] = '1';
+			*type = 1;
 		} else {
-			data[0] = '2';
+			*type = 2;
 		}
 	} else if (g_mode == RTK_WIFI_SOFT_AP_IF) {
-		data[0] = '3';
+		*type = 3;
 	}
-	argv[0] = data;
 
-	pid = kernel_thread("lwnl80211_cbk_handler", 100, 1024, (main_t)rtk_drv_callback_handler, argv);
-	if (pid < 0) {
+	ret = pthread_create(&tid, NULL, (pthread_startroutine_t)rtk_drv_callback_handler, (void *)type);

```

Comment:
pthread is for user side thread. So I think it is better to revert.

Reply:
not sure whether kernel_thread is creating the problem or not but pthread is also being used in
os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c : 694

Reply:
Dear @jeongchanKim As mentioned in above reply, linkup_handler and linkdown_handler are called many times during the stress test. The previous way to create threads may result in many cbk_handler threads be created at same time which may leads to assert. The new way to create thread ensures that only one cbk_handler is created because there's only one tid for cbk_handler thread. 
Thank you~

Reply:
@deepaksrma I think that should be changed also. As I mentioned, pthread is for user side.
@gexuyan Sorry, but I don't understand why `pthread_create` ensures only one tid for cbk_handler.
`kernel_thread` also makes one pid for each thread when creating.

Reply:
@gexuyan As I said in CC, using pthread instead of kthread can't be what you said, ' The previous way to create threads may result in many cbk_handler threads be created at same time which may leads to assert. The new way to create thread ensures that only one cbk_handler is created because there's only one tid for cbk_handler thread.`. It just leads the location change from kernel to user. In operation point of view, they are exactly the same.

Reply:
Dear @sunghan-chang , Dear @jeongchanKim ,
Ok I understood. Sorry that I will close this PR. The modifications in linkup_handler() and linkdown_handler() seems only avoid the assert issue but didn't solve the root problem. After printing out the irq number, we found this is a usage_fault in wifi msg handler thread. May I ask for developer's help to further check this section?
[irq_log.txt](https://github.com/Samsung/TizenRT/files/6047877/irq_log.txt)
![ameba_irq](https://user-images.githubusercontent.com/40012984/109265869-2e022800-7842-11eb-9a3b-f323699a6bd2.png)

Thank You~